### PR TITLE
seem to be complete

### PR DESCRIPTION
**Pull request by JustAnOtherCoder13**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -62,6 +62,20 @@ dependencies {
...
+    androidTestImplementation 'com.android.support.test:rules:1.0.2'
+    androidTestImplementation 'com.android.support.test:runner:1.0.2'
+    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
+    androidTestImplementation('com.android.support.test.espresso:espresso-contrib:3.0.2') {
+        exclude group: 'com.android.support', module: 'appcompat'
+        exclude group: 'com.android.support', module: 'support-v4'
+        exclude module: 'recyclerview-v7'
+    }
+    androidTestImplementation "android.arch.core:core-testing:1.1.1"
+    //IDLE CONCURRENT
```
Comment (fodeguirassy) [inline app/build.gradle]: can't see where you're using this

File: app/src/androidTest/java/com/openclassrooms/realestatemanager/NetworkConnectionTest.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import static com.openclassrooms.realestatemanager.presentation.utils.Utils.isInternetAvailable;
+import static org.junit.Assert.assertTrue;
+
+@LargeTest
+@RunWith(AndroidJUnit4.class)
+public class NetworkConnectionTest {
+
+    @Rule
+    public InstantTaskExecutorRule instantTaskExecutorRule = new InstantTaskExecutorRule();
+
```
Comment (fodeguirassy) [inline app/src/androidTest/java/com/openclassrooms/realestatemanager/NetworkConnectionTest.java:29]: I tough you'd enable an disable internet connection. Then test wether your Utils function behaves as expected.

**Review by fodeguirassy — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/com/openclassrooms/realestatemanager/NetworkConnectionTest.java (hunk 2 above)
Comment (JustAnOtherCoder13) [reply app/src/androidTest/java/com/openclassrooms/realestatemanager/NetworkConnectionTest.java:29]: I've seen no way to enable disable connection for data and wifi programatically

File: app/build.gradle (hunk 1 above)
Comment (JustAnOtherCoder13) [reply app/build.gradle]: you're right, done

File: app/src/androidTest/java/com/openclassrooms/realestatemanager/NetworkConnectionTest.java (hunk 2 above)
Comment (fodeguirassy) [reply app/src/androidTest/java/com/openclassrooms/realestatemanager/NetworkConnectionTest.java:29]: How about [https://stackoverflow.com/questions/56741122/unit-test-for-network-utility-class](url) ?

Comment (JustAnOtherCoder13) [reply app/src/androidTest/java/com/openclassrooms/realestatemanager/NetworkConnectionTest.java:29]: i'me gonna test that

Comment (JustAnOtherCoder13) [reply app/src/androidTest/java/com/openclassrooms/realestatemanager/NetworkConnectionTest.java:29]: Capture


That's in what you've send me, if i understand well, after API17, every try to access phone data will fail cause protected.

I also couldn't disable wifi on test never pass :

assertNotEquals(networkInfo.getType(), ConnectivityManager.TYPE_WIFI);

neither more :

        assertFalse(wifiManager.isWifiEnabled());

Comment (fodeguirassy) [reply app/src/androidTest/java/com/openclassrooms/realestatemanager/NetworkConnectionTest.java:29]: I was thinking about something like

`
@Test
public void isWifiEnabled() {
    WifiManager wifiManager = (WifiManager) mContext.getSystemService(Context.WIFI_SERVICE);
    wifiManager.setWifiEnabled(false);
    wifiManager.disconnect();

    assertFalse(NetworkUtils.isConnectedWifi(mContext));
}
`

**Pull request merged**